test/should_pass/intersection_with_any.erl

```diff
@@ -7,9 +7,9 @@
 -spec any_refined_using_guard(any()) -> 5.
 any_refined_using_guard(X) when is_integer(X) -> X.
 
--spec intersection_using_constraints(X) when X :: integer(),
-                                             X :: any() ->
-                                                       5.
+-spec intersection_using_constraints(X) -> X when
+                                                X :: integer(),
+                                                X :: any().
```

Comment:
Wow, amazing :)

Reply:
But the return value should be `5`, i.e. a subset of `integer()`. It should fail if `X :: any()` is removed, since X :: integer() would mean any integer but `integer() & any()` would mean "some unknown subset of integer". At least I think that was the intention of this originally, but it's hard to tell since it was a syntax error.

Reply:
Yep, - there are two tests now. - this one (which is testing the happy path) and that one:

https://github.com/josefs/Gradualizer/blob/d86b6b54b321255255afe26b80153fa2ea5dd81d/test/should_fail/intersection_with_any.erl#L5-L8

I believe that both intentions are there.

Reply:
Note `should_pass` vs `should_fail` locations

Reply:
Ah, thanks. Then, I think `(X) -> 5 when X :: integer(), X :: any()` should be under "known_problems/should_pass".

`X :: any()` makes it a dynamic ( = unknown at compile time) type which combined with `X :: integer()` makes it an unknown subtype of `integer()`, which may well be the singleton type `5`, so by gradual typing, we shouldn't give an error here.

We had some discussion about intersection types in #130. See also the paper *G. Castagna and V. Lanvin: Gradual Typing with Union and Intersection Types* [icfp17.pdf](https://www.irif.fr/~gc/papers/icfp17.pdf).

Reply:
Got it,- I mixed/confused `any()` with `term()` here. Feel free to move it.

(Changing `any()` -> `term()` - makes it fine to me, and now I understand/capture this subtlety).

Reply:
Actually `any()` and `term()` are both the dynamic type in Gradualizer. Originally `term()` meant the union of all static types, but we changed it because of how `term()` is used in practice in OTP and elsewhere.

Reply:
> so by gradual typing, we shouldn't give an error here

I see: https://github.com/josefs/Gradualizer/pull/305/commits/bf290481fb788af9ab3a4d8932173470bd5bc577

Reply:
Justice restored: #305